include/zephyr/location.h

```diff
@@ -0,0 +1,402 @@
+/**
+ * @file location.h
+ *
+ * @brief Public location API.
+ *
+ */
+
+/*
+ * Copyright (c) 2022, Bjarki Arge Andreasen
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_LOCATION_H_
+#define ZEPHYR_INCLUDE_LOCATION_H_
+
+/**
+ * @brief Location Interface
+ * @defgroup location_interface Location Interface
+ * @ingroup io_interfaces
+ * @{
+ */
+
+#include <zephyr/types.h>
+#include <zephyr/device.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Location data
+ *
+ * The latitude and longitude are in the unit micro degree. 1 micro
+ * degree corresponds to approx. 11cm along the earths surface.
```

Comment:
Only true at the equator and along the meridians.

Reply:
Yes, I will update the description to be more accurate once we determine if we go for nano degrees or microdegrees :)

Reply:
Updated description for microdegrees for now